networkx/algorithms/operators/binary.py

```diff
@@ -170,7 +170,7 @@ def intersection(G, H):
     return nx.intersection_all([G, H])
 
 
-@nx._dispatch(graphs=_G_H)
+@nx._dispatch(graphs=_G_H, preserve_node_attrs=True, preserve_graph_attrs={"G"})
```

Comment:
So, I'm not sure if the "Notes" section about `difference` is correct. Both graph and node attributes are copied over from G to the new graph. This could probably because `nx.create_empty_copy` below _is not_ called with `with_data=False`.

I updated the dispatch decorator to be consistent with observed behavior, but this means it's different than how the docstring says it should behave.

Reply:
I think we should update the doc_string to reflect what we actually do. We do not copy the edge attributes. The workaround displayed allows the user to copy the edge attributes. 

We could also change the function to use `with_data=False`. But I guess that requires a deprecation (not sure whether deprecation required to make function match the docs, but probably). 

Ack -- this also is true of `symmetric_difference`. The docs say the node attributes aren't copied, but they are. 

Does anyone think we need `difference` on graphs to create a slim version without node or graph attributes?